src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -607,32 +607,17 @@ void network::allocate_primitives() {
                     return (lhs->get_output_layout().bytes_count() > rhs->get_output_layout().bytes_count());
               });
 
+    // Move layers that can be optimized to the back of nodes_to_allocate
```

Comment:
Could you follow-up by stating "why" instead of "what"?

Reply:
The layers to be optimized use the input buffer as the output buffer. These layers must be allocated after the input/output buffer of each previous layer is allocated. For this, it is necessary to send the layers to be optimized back to be processed last.